airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/generate_erd/pipeline.py

```diff
@@ -0,0 +1,176 @@
+        super().__init__(context)
+        self._skip_relationship_generation = skip_relationship_generation
+
+    async def _run(self, connector_to_discover: Container) -> StepResult:
+        if not self._skip_relationship_generation and not self.context.genai_api_key:
+            raise ValueError("GENAI_API_KEY needs to be provided if the relationship generation is not skipped")
+
+        discovered_catalog = await self._get_discovered_catalog(connector_to_discover)
+
+        connector_directory = await self.context.get_connector_dir()
+        command = ["poetry", "run", "erd", "--source-path", "/source"]
+        if self._skip_relationship_generation:
+            command.append("--skip-llm-relationships")
+
+        await (
+            self._build_erd_container(connector_directory, discovered_catalog)
+            .with_exec(command)
+            .directory("/source/erd")
+            .export(str(_get_erd_folder(self.context.connector.code_directory)))
+        )
+
+        return StepResult(step=self, status=StepStatus.SUCCESS)
+
+    async def _get_discovered_catalog(self, connector_to_discover: Container) -> str:
+        source_config_path_in_container = "/data/config.json"
+        discover_output = (
+            await connector_to_discover.with_new_file(source_config_path_in_container, contents=self._get_default_config().value)
+            .with_exec(["discover", "--config", source_config_path_in_container])
+            .stdout()
+        )
+        return self._get_schema_from_discover_output(discover_output)
+
+    def _get_default_config(self) -> Secret:
+        if not self.context.local_secret_store:
+            raise ValueError("Expecting local secret store to be set up but couldn't find it")
+
+        filtered_secrets = [secret for secret in self.context.local_secret_store.get_all_secrets() if secret.file_name == "config.json"]
+        if not filtered_secrets:
+            raise ValueError("Expecting at least one secret to match name `config.json`")
+        elif len(filtered_secrets) > 1:
+            print(f"Expecting only one secret with name `config.json but got {len(filtered_secrets)}. Will take one randomly...")
+
+        return filtered_secrets[0]
+
+    @staticmethod
+    def _get_schema_from_discover_output(discover_output: str) -> str:
+        for line in discover_output.split("\n"):
+            json_line = json.loads(line)
+            if json_line.get("type") == "CATALOG":
+                return json.dumps(json.loads(line).get("catalog"))
+        raise ValueError("No catalog was found in output")
+
+    def _build_erd_container(self, connector_directory: Directory, discovered_catalog: str) -> Container:
+        """Create a container to run ERD generation."""
+        container = with_poetry(self.context)
+        if self.context.genai_api_key:
+            container = container.with_env_variable("GENAI_API_KEY", self.context.genai_api_key.value)
```

Comment:
```suggestion
            container = container.with_secret_variable("GENAI_API_KEY", self.context.genai_api_key.as_dagger_secret())
```
This will make sure dagger does not print out the secret